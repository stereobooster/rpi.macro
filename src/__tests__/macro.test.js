const path = require("path");
const pluginTester = require("babel-plugin-tester");
const plugin = require("babel-plugin-macros");
const prettier = require("prettier");

pluginTester({
  plugin,
  snapshot: true,
  babelOptions: {
    filename: __filename,
  },
  formatResult(result) {
    return prettier.format(result, { trailingComma: "es5" });
  },
  tests: {
    "no usage": `import rpi from '../macro'`,
    "correct usage": `
      import rpi from '../macro';

      const preview = rpi('./fixtures/jonathan-daniels-416786-unsplash.jpg');
    `,
    // "correct usage, wrong file": `
    //   import rpi from '../macro';

    //   const preview = rpi('./fixtures/readme.md');
    // `,
  },
});
