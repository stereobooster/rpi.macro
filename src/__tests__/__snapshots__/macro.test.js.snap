// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`macros correct usage: correct usage 1`] = `
"
import rpi from '../macro';

const preview = rpi('./fixtures/jonathan-daniels-416786-unsplash.jpg');

      ↓ ↓ ↓ ↓ ↓ ↓

const preview = {
  placeholder: {
    lqip:
      \\"data:image/jpeg;base64,/9j/2wBDAAYEBQYFBAYGBQYHBwYIChAKCgkJChQODwwQFxQYGBcUFhYaHSUfGhsjHBYWICwgIyYnKSopGR8tMC0oMCUoKSj/2wBDAQcHBwoIChMKChMoGhYaKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCgoKCj/wAARCAAJAA4DASIAAhEBAxEB/8QAFgABAQEAAAAAAAAAAAAAAAAAAwYH/8QAJxAAAQMCBQIHAAAAAAAAAAAAAQIDBAAFBhESITEHsTQ2QUJyc4H/xAAVAQEBAAAAAAAAAAAAAAAAAAABAv/EABgRAAMBAQAAAAAAAAAAAAAAAAABAhEh/9oADAMBAAIRAxEAPwCB6ZxbDcHn2MUGSvVk3GZbUW06iFHUpQzIyyAGx54ocdYMuWHC3IS067ZpSyYj5IGsb7KHtUB6fop5vio/yHcVrV98m2/7R2XS+UQpTnD/2Q==\\",
  },
  width: 3024,
  height: 4774,
  size: 436854,
};

"
`;

exports[`macros no usage: no usage 1`] = `
"
import rpi from '../macro'

      ↓ ↓ ↓ ↓ ↓ ↓


"
`;
